src/schema/config.ts

```diff
@@ -89,7 +89,7 @@ export interface ApiClientConfig {
   }
   client: {
     client_id: string
-    client_secret: string
+    client_secret?: string
```

Comment:
yes, I think that your are right, perhaps we could add a comment here that client secret is only necessary when using client_secret_basic auth